application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/producer/ExchangeProducerBean.java

```diff
@@ -10,24 +10,41 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
 
 package eu.europa.ec.fisheries.uvms.activity.message.producer;
 
+import eu.europa.ec.fisheries.schema.exchange.module.v1.EfrReportSaved;
+import eu.europa.ec.fisheries.schema.exchange.module.v1.ExchangeModuleMethod;
 import eu.europa.ec.fisheries.uvms.commons.message.api.MessageConstants;
 import eu.europa.ec.fisheries.uvms.commons.message.impl.AbstractProducer;
+import eu.europa.ec.fisheries.uvms.exchange.model.mapper.JAXBMarshaller;
 
 import javax.annotation.Resource;
 import javax.ejb.LocalBean;
 import javax.ejb.Stateless;
 import javax.jms.Destination;
+import javax.jms.JMSException;
 import javax.jms.Queue;
+import java.util.UUID;
 
 @Stateless
 @LocalBean
 public class ExchangeProducerBean extends AbstractProducer {
 
-    @Resource(mappedName =  "java:/" + MessageConstants.QUEUE_EXCHANGE_EVENT)
+    @Resource(mappedName = "java:/" + MessageConstants.QUEUE_EXCHANGE_EVENT)
     private Queue destination;
 
     @Override
     public Destination getDestination() {
         return destination;
     }
+
+    public void sendEfrReportSavedAckToExchange(UUID ackResponseMessageId) throws JMSException {
+        if (ackResponseMessageId == null) {
+            return;
+        }
+        EfrReportSaved efrReportSaved = new EfrReportSaved();
+        efrReportSaved.setMethod(ExchangeModuleMethod.EFR_REPORT_SAVED);
+        efrReportSaved.setRefGuid(ackResponseMessageId.toString());
+
+        String xml = JAXBMarshaller.marshallJaxBObjectToString(efrReportSaved);
```

Comment:
Don't we want the ACK response to be JSON?

Reply:
The EfrReportSaved-message is defined in an xml schema in exchange-model and I think Exchange expects it to be xml only, in order to parse and save it